scripts/weihergames_data_extractor.py

```diff
@@ -0,0 +1,212 @@
+    pass
+    # set player:
+    for name, data in j['members'].items():
+        p = PlayerData(name)
+        p.elo = ELO_START_VALUE
+        p.img = data.get('img', '')
+        players.append(p)
+        logging.debug('Player found in DB: {0}'.format(p))
+    return players
+
+
+def read_games_from_json():
+    games = []
+    # Load data
+    j = json.load(open(JSON_DATABASE))
+    # Sanity check of json file:
+    pass
+    #output all warnings + errors, if no error continue:
+    pass
+    # set player:
+    for name, data in j['games'].items():
+        g = GameData(name)
+        g.img = data.get('img', '')
+        g.field_of_play = data.get('field_of_play', '')
+        g.field_of_play_icon = data.get('field_of_play_icon', '')
+        g.pointrules = data.get('pointrules', [])
+        g.specific_leveling_value = data.get('specific_leveling_value', 0)
+        games.append(g)
+        logging.debug('Game found in DB: {0}'.format(g))
+    return games
+
+
+def read_matches_from_json(players, games):
+    matches = []
+    # Load data
+    j = json.load(open(JSON_DATABASE))
+    # sort matches by date:
+    sorted_matchlist_db = sorted(j['matches'], key=itemgetter('time'))
+    # go through all matches and iterate ELO:
+    for match_db in sorted_matchlist_db:
+        # search for the game played:
+        game_db = match_db['game']
+        g = next((x for x in games if x.name == game_db), None)
+        if g is None:
+            logging.warning('GameData not in database: {0}'.format(game_db))
+            g = GameData(game_db)
+            games.append(g)
+        m = MatchData(g)
+        for participant_db, points_db in match_db['result'].items():
+            # search for player with same name:
+            p = next((x for x in players if x.name == participant_db), None)
+            if p is None:
+                logging.error('Matchdata has playername not in database: {0}'.format(participant_db))
+                p = PlayerData(participant_db)
+                players.append(p)
+            # keep same order for particpants and points lists!!!:
+            m.participants.append(p)
+            m.result.append(points_db)
+            m.location = match_db.get('location', '')
+            m.remarks = match_db.get('remarks', '')
+            m.datetime = match_db.get('datetime', '')
+        matches.append(m)
+        logging.debug('Match appended: {0}'.format(m))
+    return matches
+
+def process_matches(matches):
+    pass # TODO
+
+
+if __name__ == '__main__':
+    logging.info('RUNNING DATA EXTRACTOR')
+    players = read_player_from_json()
+    games = read_games_from_json()
+    matches = read_matches_from_json(players, games)
+    process_matches(matches)
+
+    templates_dir = os.path.join(_ROOT_FILE, '../templates')
+    env = Environment(
+        loader=FileSystemLoader(templates_dir),
+        autoescape=select_autoescape(['html', 'xml'])
+    )
+
+    template = env.get_template('test.html')
```

Comment:
please add some utils to the template
https://stackoverflow.com/questions/30104914/how-to-get-current-time-in-jinja

`getYearFromDate(dateString) => number`
`removeUser(array, username/userid) => array without the corresponding user`

Reply:
getYearFromDateString its called

Reply:
please test